pkg/backup/restore.go

```diff
@@ -69,9 +71,21 @@ func Restore(cfg *config.Config, backupName, tablePattern string, databaseMappin
 		}
 		if schemaOnly || doRestoreData {
 			for _, database := range backupMetadata.Databases {
-				if !IsInformationSchema(database.Name) {
-					if err := ch.CreateDatabaseFromQuery(database.Query); err != nil {
-						return err
+				if targetDB, ok := dbMapRule[database.Name]; ok && len(dbMapRule) > 0 {
+					// When create database, try to substitute the database by following the database mapping rule.
+					if !IsInformationSchema(targetDB) {
+						regExp := regexp.MustCompile(`(?m)^CREATE DATABASE ([\x60]?)([^\x60]*)([\x60]?)`)
```

Comment:
I think we could move this compilation into `pkg/clickhouse/clickhouse.go`, something like `var createDatabaseRE = regexp.MustCompile()` 

and use it as `clickhouse.createDatabaseRE.ReplaceAllString(...)` in code